test/terraform_remote_exec_example_test.go

```diff
@@ -0,0 +1,109 @@
+package test
+
+import (
+	"fmt"
+	"github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/ssh"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/gruntwork-io/terratest/modules/test-structure"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+// This test shows how to override the systems local SSH Agent, with an in-process SSH agent, whose keys can be managed
+// from within your tests. This allows you to test Terraform modules which make SSH connections to the created
+// instances, useful for tasks such as provisioning.
+func TestTerraformRemoteExecExample(t *testing.T) {
+	t.Parallel()
+
+	terraformDirectory := "../examples/terraform-remote-exec-example"
+
+	// Deploy the example
+	test_structure.RunTestStage(t, "setup", func() {
+
+		// A unique ID we can use to namespace resources so we don't clash with anything already in the AWS account or
+		// tests running in parallel
+		uniqueID := random.UniqueId()
+
+		// Give this EC2 Instance and other resources in the Terraform code a name with a unique ID so it doesn't clash
+		// with anything else in the AWS account.
+		instanceName := fmt.Sprintf("terratest-remote-exec-example-%s", uniqueID)
+
+		// Pick a random AWS region to test in. This helps ensure your code works in all regions.
+		awsRegion := aws.GetRandomRegion(t, nil, nil)
+
+		// Create an EC2 KeyPair that we can use for SSH access
+		keyPairName := fmt.Sprintf("terratest-remote-exec-example-%s", uniqueID)
+		keyPair := aws.CreateAndImportEC2KeyPair(t, awsRegion, keyPairName)
```

Comment:
`defer` deleting the KeyPair.

Reply:
This was already in the tearDown step that I just moved from the bottom. Let me know if you meant something else.